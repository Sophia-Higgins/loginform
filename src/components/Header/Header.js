import React from "react";
import { useHistory, useLocation } from "react-router-dom";
import { ACCESS_TOKEN_NAME } from "../../constants/apiConstants";

function Header() {
  let history = useHistory();
  let location = useLocation();

  function renderLogout() {
    if (location.pathname === "/home") {
      return (
        <div>
          <button className="btn btn-light" onClick={() => handleLogout()}>
            Logout
          </button>
        </div>
      );
    }
  }

  function handleLogout() {
    localStorage.removeItem(ACCESS_TOKEN_NAME);
    history.push("/login");
  }

  return (
    <nav className="navbar navbar-dark bg-primary">
      <div className="col-sm" />
      <div className="row col-sm d-flex justify-content-center text-white">
        <span className="h3">The Good Place</span>
      </div>
      <div className="row col-sm d-flex justify-content-end">
        {renderLogout()}
      </div>
    </nav>
  );
}
export default Header;
