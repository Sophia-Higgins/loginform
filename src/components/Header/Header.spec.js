import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import Header from './Header';

let wrapper;

beforeEach(() => {
  wrapper = mount(
    <MemoryRouter>
      <Header />
    </MemoryRouter>
  );
});

describe('<Header />', () => {
  it('should match the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
