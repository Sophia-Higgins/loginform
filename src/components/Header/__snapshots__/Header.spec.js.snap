// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match the snapshot 1`] = `
<nav
  className="navbar navbar-dark bg-primary"
>
  <div
    className="col-sm"
  />
  <div
    className="row col-sm d-flex justify-content-center text-white"
  >
    <span
      className="h3"
    >
      The Good Place
    </span>
  </div>
  <div
    className="row col-sm d-flex justify-content-end"
  />
</nav>
`;
